### PR TITLE
Complete Ticket 10

### PR DESCRIPTION
**Pull request by Brynster99**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -118,6 +157,26 @@ describe('GET /api/articles/:article_id', () => {
...
+        expect(article).toEqual(
+          expect.objectContaining({
+            article_id: 1,
+            title: 'Living in the shadow of a great man',
+            topic: 'mitch',
+            author: 'butter_bridge',
+            body: 'I find this existence challenging',
+            created_at: '2020-07-09T20:11:00.000Z',
+            votes: 100,
+            comment_count: '11',
```
Comment (daviidbartlett) [inline __tests__/endpoints.test.js:168]: this is the only new bit to the object, no? Only need to test the property I reckon

File: controllers/controllers.js [hunk 2]
```diff
@@ -28,7 +37,13 @@ exports.getArticles = (req, res, next) => {
```
Comment (daviidbartlett) [inline controllers/controllers.js]: remove console.log ☝️

File: controllers/controllers.js [hunk 3]
```diff
@@ -28,7 +37,13 @@ exports.getArticles = (req, res, next) => {
 
   fetchArticles()
     .then((articles) => {
-      res.status(200).send({ articles });
+      const articlePromises = articles.map((article) => {
+        return fetchArticleById(article.article_id).then((article) => article);
+      });
+
+      Promise.all(articlePromises).then((articles) => {
```
Comment (daviidbartlett) [inline controllers/controllers.js]: why are you doing any of this? `fetchArticles` got you some article and you're sending them back, no? Why are you fetching article by ID afterwards?

File: controllers/controllers.js [hunk 4]
```diff
@@ -46,7 +61,6 @@ exports.getArticleById = (req, res, next) => {
 exports.patchArticleById = (req, res, next) => {
   console.log('invoked patchArticleById');
```
Comment (daviidbartlett) [inline controllers/controllers.js]: remove

File: models/models.js [hunk 5]
```diff
@@ -25,7 +34,10 @@ exports.fetchArticleById = (articleId) => {
   console.log('invoked fetchArticleById');
```
Comment (daviidbartlett) [inline models/models.js]: remove

File: models/models.js [hunk 6]
```diff
@@ -1,6 +1,15 @@
...
+exports.fetchApiDocs = () => {
+  console.log('invoked fetchApiDocs');
+
+  return fs
+    .readFile(`${__dirname}/../endpoints.json`)
+    .then((fileContent) => JSON.parse(fileContent));
+};
+
 exports.fetchUsers = () => {
   console.log('invoked fetchUsers');
```
Comment (daviidbartlett) [inline models/models.js]: remove

File: __tests__/endpoints.test.js [hunk 7]
```diff
@@ -124,13 +124,7 @@ describe('GET /api/articles', () => {
...
         expect(articles[0]).toEqual(
           expect.objectContaining({
-            author: expect.any(String),
-            title: expect.any(String),
-            article_id: expect.any(Number),
-            topic: expect.any(String),
-            created_at: expect.any(String),
-            votes: expect.any(Number),
-            comment_count: expect.any(String),
+            comment_count: '2',
```
Comment (dav3rid) [inline __tests__/endpoints.test.js:127]: 👍

File: models/models.js [hunk 8]
```diff
@@ -3,36 +3,28 @@ const fs = require('fs/promises');
...
   return db.query('SELECT * FROM topics;').then(({ rows }) => rows);
 };
 
 exports.fetchArticles = () => {
-  console.log('invoked fetchArticles');
-
   return db
-    .query('SELECT * FROM articles ORDER BY created_at DESC;')
+    .query(
+      'SELECT articles.*, COUNT(articles.article_id) as comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;'
```
Comment (dav3rid) [inline models/models.js:22]: Well done

**Pull request merged**